src/services/DataService.ts

```diff
@@ -0,0 +1,26 @@
+import axios from 'axios';
+
+import { type SensorData } from '@/mocks/SensorData';
+import { AirImpurityDataModel } from '@/models/AirImpurityDataModel';
+import { SensorDataModel } from '@/models/SensorDataModel';
+
+const locationsURI = `${import.meta.env.VITE_LOCAL_API as string}/sampleData`;
+export default async function getRecent() {
+  const res = await axios.get(locationsURI);
+  if (res.status) {
+    const resJSON: [
+      { id: string; deviceLocation: string; pm10Quantity: number }
+    ] = res.data;
+    const sampleData: [SensorData] = resJSON.map(
+      value =>
+        new SensorDataModel(
+          value.deviceLocation,
+          [new AirImpurityDataModel('pm10Qunatity', value.pm10Quantity)],
+          new Date()
+        )
+    );
+    return sampleData;
+  }
+  console.log('Failed to fetch locations.');
+  return [];
+}
```

Comment:
```suggestion
import axios from 'axios';

import { type SensorData } from '@/mocks/SensorData';
import { type MeasurementsResponse } from '@/types/api/measurements';

const measurementsURI = `${
  import.meta.env.VITE_LOCAL_API as string
}/sample-data`;

export const getMeasurements = async (): Promise<SensorData[]> => {
  try {
    const res = await axios.get<MeasurementsResponse>(measurementsURI);
    return res.data.map(value => ({
      code: value.id,
      date: new Date(value.date),
      data: [
        {
          name: 'PM10',
          value: value.pm10Quantity
        }
      ]
    }));
  } catch (error) {
    console.error(error);
  }
  return [];
};
```

And add these types to the `src/types/api/measurements.ts` file:
```ts
export interface MeasurementDTO {
  id: string;
  date: string;
  deviceLocation: string;
  pm10Quantity: number;
}

export type MeasurementsResponse = MeasurementDTO[];
```

Also add `date` field in [this](https://github.com/AtmosFears/AtmosFears-BE/pull/14) to the object returned from the backend because it is a measurement date, not the current date.

Change the endpoint name from `sampleData` to `sample-data` - kebab case is preferred for api endpoints (alternatively, you can use the snake case if you wish).

I also don't understand why you return `deviceLocation` and you create a separate endpoint to fetch just locations of devices. I think you should fix backend implementation first and then make changes on the frontend as I don't understand what is your intention in here.